src/Particle.js

```diff
@@ -23,8 +23,40 @@ class Particle {
 	 * @param  {Object} options Options to be used for all requests (see [Defaults](../src/Defaults.js))
 	 */
 	constructor(options = {}) {
+		// todo - this seems a bit dangerous - would be better to put all options/context in a contained object
 		Object.assign(this, Defaults, options);
+		this.context = {};
 		this.agent = new Agent(this.baseUrl);
+		this.setContext({name:'tool', value: this.tool});
+		this.setContext({name:'project', value: this.project});
+	}
+
+	_isValidContext(name, context) {
+		return (name==='tool' || name==='project') && context!==undefined;
+	}
+
+	setContext(name, context) {
+		if (context!==undefined) {
+			if (this._isValidContext(name, context)) {
+				this.context[name] = context;
+			} else {
+				throw Error('uknown context name or undefined context: '+name);
```

Comment:
This does not throw for undefined context even though the error text suggests it should.